src/etc/rustup.sh

```diff
@@ -453,16 +453,33 @@ then
     RUST_URL="${RUST_URL}/${CFG_DATE}"
 fi
 
-verify_hash() {
-    remote_sha256="$1"
-    local_file="$2"
-
+download_hash() {
     msg "Downloading ${remote_sha256}"
     remote_sha256=`"${CFG_CURL}" -f "${remote_sha256}"`
+    if [ -n "{CFG_SAVE}" ]; then
+        echo "${remote_sha256}" > "${local_sha_file}"
+    fi
     if [ "$?" -ne 0 ]; then
         rm -Rf "${CFG_TMP_DIR}"
         err "Failed to download ${remote_url}"
     fi
+}
+
+verify_hash() {
+    remote_sha256="$1"
+    local_file="$2"
+    local_sha_file="${local_file}.sha256"
+
+    if [ -n "${CFG_SAVE}" ]; then
```

Comment:
Not sure, currently `download_hash` can be called when `${CFG_SAVE}` is both specified or not specified.
In case of specified, if file exists, we use that. If file doesn't exist, we download it to compare with, and save for future use.
In case `${CFG_SAVE}` is specified, we download the file anyway and don't save it since it's not requested.
Simplifying to suggested variant would mean we check existence of saved file when `${CFG_SAVE}` wasn't specified at all. I believe we shouldn't look at local files when saving wasn't requested.